schema/crdb/dbinit.sql

```diff
@@ -242,7 +242,13 @@ CREATE TABLE IF NOT EXISTS omicron.public.sled_resource (
     reservoir_ram INT8 NOT NULL,
 
     -- Identifies the type of the resource
-    kind omicron.public.sled_resource_kind NOT NULL
+    kind omicron.public.sled_resource_kind NOT NULL,
+
+    -- The UUID of an instance, if this resource belongs to an instance.
+    instance_id UUID
```

Comment:
I assume this is being added so that the `sled_resource` table can also be used to track which instances are present on a sled? But, this information is already present in the `vmm` table, where records have both the sled ID of the VMM that the sled is on and the ID of the instance that is running on that VMM. For my education,[^1] would you mind explaining what putting instance IDs here accomplishes?

[^1]: Don't take this as me saying I don't like adding this, I'm just curious about what it accomplishes in the overall design. :)

Reply:
It's a good question: https://github.com/oxidecomputer/omicron/pull/7076#discussion_r1933125385

The TL;DR is that "the VMM table is not concurrently created with the resource table" - it is possible to create a reservation on behalf of an instance without necessarily creating a VMM record.